src/iter.c

```diff
@@ -841,6 +841,8 @@ ecs_iter_t ecs_page_iter(
     ecs_check(it->next != NULL, ECS_INVALID_PARAMETER, NULL);
 
     ecs_iter_t result = *it;
+    result.priv.cache.stack_cursor = NULL; /* Don't copy allocator cursor */
```

Comment:
Yes. These were the places where I ran into issues with storing pointers. Glad you had the context to know what was expected of the wrapper iterator!